### PR TITLE
Add Nginx, Redis Postgress and soc listener plugins (AO-16696)

### PR DESCRIPTION
**Pull request by d-maslyk**
* AO-16696
*Add Nginx, Nginx Plus, nginx Plus API, Redis, Postgress and socket listener plugins - with basic params

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-DOCKERFILE_VERSION=3.3.0
+DOCKERFILE_VERSION=3.3.1
```
Comment (melord) [inline Makefile:1]: I think 3.4.0, we are adding functionality

File: Makefile [hunk 2]
```diff
@@ -12,7 +12,7 @@ build:
 
 .PHONY: build-test
 build-test: 
-	@docker build -t $(CURRENT_IMAGE) -t $(LATEST_IMAGE) --build-arg swisnap_repo=swisnap --build-arg swisnap_version=$(SWISNAP_VERSION) .
+	@docker build -t $(CURRENT_IMAGE) -t $(LATEST_IMAGE) --build-arg swisnap_repo=swisnap-stg --build-arg swisnap_version=$(SWISNAP_VERSION) .
```
Comment (melord) [inline Makefile:15]: why staging repo?

File: README.md [hunk 3]
```diff
@@ -192,11 +192,17 @@ The following environment parameters are available:
  SWISNAP_ENABLE_APACHE          | Set this to `true` to enable the Apache plugin.
  SWISNAP_ENABLE_ELASTICSEARCH   | Set this to `true` to enable the Elasticsearch plugin.
  SWISNAP_ENABLE_KUBERNETES      | Set this to `true` to enable the Kubernetes plugin. Enabling this option on the DaemonSet will cause replication of Kubernetes metrics where the replication count is the number of pods with Kubernetes collection enabled minus one. Typically Kubernetes collection is only enabled on the Deployment asset.
+ SWISNAP_ENABLE_NGINX           | Set this to `true` to enable the Nginx Bridge. If enabled the following ENV vars are required to be set:<br>*NGINX_STATUS_URI* - one, or multiple space separeted link(s) to Nginx stub_status URI.
```
Comment (melord) [inline README.md:195]:
```suggestion
 SWISNAP_ENABLE_NGINX           | Set this to `true` to enable the Nginx Bridge plugin. If enabled the following ENV vars are required to be set:<br>*NGINX_STATUS_URI* - one, or multiple space separeted link(s) to Nginx stub_status URI.
```

File: README.md [hunk 4]
```diff
@@ -192,11 +192,17 @@ The following environment parameters are available:
...
+ SWISNAP_ENABLE_NGINX_PLUS      | Set this to `true` to enable the Nginx Plus Bridge. If enabled the following ENV vars are required to be set:<br>*NGINX_PLUS_STATUS_URI* - one, or multiple space separeted link(s) to ngx_http_status_module or status URI.
```
Comment (melord) [inline README.md:196]:
```suggestion
 SWISNAP_ENABLE_NGINX_PLUS      | Set this to `true` to enable the Nginx Plus Bridge plugin. If enabled the following ENV vars are required to be set:<br>*NGINX_PLUS_STATUS_URI* - one, or multiple space separeted link(s) to ngx_http_status_module or status URI.
```

File: README.md (hunk 3 above)
Comment (lootek) [inline README.md:195]:
```suggestion
 SWISNAP_ENABLE_NGINX           | Set this to `true` to enable the Nginx plugin. If enabled the following ENV vars are required to be set:<br>*NGINX_STATUS_URI* - one, or multiple space-separated link(s) to Nginx stub_status URI.
```

File: README.md (hunk 4 above)
Comment (lootek) [inline README.md:196]:
```suggestion
 SWISNAP_ENABLE_NGINX_PLUS      | Set this to `true` to enable the Nginx Plus plugin. If enabled the following ENV vars are required to be set:<br>*NGINX_PLUS_STATUS_URI* - one, or multiple space-separated link(s) to ngx_http_status_module or status URI.
```

File: README.md [hunk 5]
```diff
@@ -192,11 +192,17 @@ The following environment parameters are available:
...
+ SWISNAP_ENABLE_NGINX_PLUS_API  | Set this to `true` to enable the Nginx Plus Api Bridge. If enabled the following ENV vars are required to be set:<br>*NGINX_PLUS_STATUS_URI* - one, or multiple space separeted link(s) to Nginx API URI.
```
Comment (lootek) [inline README.md:197]:
```suggestion
 SWISNAP_ENABLE_NGINX_PLUS_API  | Set this to `true` to enable the Nginx Plus API plugin. If enabled the following ENV vars are required to be set:<br>*NGINX_PLUS_STATUS_URI* - one, or multiple space-separated link(s) to Nginx API URI.
```

File: README.md [hunk 6]
```diff
@@ -192,11 +192,17 @@ The following environment parameters are available:
...
  SWISNAP_ENABLE_MESOS           | Set this to `true` to enable the Mesos plugin.
  SWISNAP_ENABLE_MONGODB         | Set this to `true` to enable the MongoDB plugin.
- SWISNAP_ENABLE_MYSQL           | Set this to `true` to enable the MySQL plugin. If enabled the following ENV vars are required to be set as well: MYSQL_USER, MYSQL_PASS, MYSQL_HOST & MYSQL_PORT
- SWISNAP_ENABLE_PROMETHEUS      | Set this to `true` to enable prometheus pod annotation scrapping
+ SWISNAP_ENABLE_MYSQL           | Set this to `true` to enable the MySQL plugin. If enabled the following ENV vars are required to be set:<br>*MYSQL_USER*,<br>*MYSQL_PASS*,<br>*MYSQL_HOST*<br>*MYSQL_PORT*
+ SWISNAP_ENABLE_PROMETHEUS      | Set this to `true` to enable Prometheus pod annotation scrapping
```
Comment (lootek) [inline README.md:201]:
```suggestion
 SWISNAP_ENABLE_PROMETHEUS      | Set this to `true` to enable the Prometheus pod annotation scrapping
```

File: README.md [hunk 7]
```diff
@@ -192,11 +192,17 @@ The following environment parameters are available:
...
+ SWISNAP_ENABLE_POSTGRESQL      | Set this to `true` to enable the Postgres plugin.  If enabled the following ENV vars are required to be set:<br>*POSTGRES_ADDRESS* - specify address for Postgres databse
  SWISNAP_ENABLE_RABBITMQ        | Set this to `true` to enable the RabbitMQ plugin.
+ SWISNAP_ENABLE_REDIS           | Set this to `true` to enable the Redis plugin. If enabled the following ENV vars are required to be set:<br>*REDIS_SERVERS* - one, or multiple space separeted link(s) to Redis servers
```
Comment (lootek) [inline README.md:204]:
```suggestion
 SWISNAP_ENABLE_REDIS           | Set this to `true` to enable the Redis plugin. If enabled the following ENV vars are required to be set:<br>*REDIS_SERVERS* - one, or multiple space-separated link(s) to Redis servers
```

File: README.md (hunk 5 above)
Comment (melord) [inline README.md:197]:
```suggestion
 SWISNAP_ENABLE_NGINX_PLUS_API  | Set this to `true` to enable the Nginx Plus Bridge plugin with Api option. If enabled the following ENV vars are required to be set:<br>*NGINX_PLUS_STATUS_URI* - one, or multiple space separeted link(s) to Nginx API URI.
```

File: deploy/overlays/stable/daemonset/kustomization.yaml [hunk 8]
```diff
@@ -8,4 +8,4 @@ bases:
 
 images:
   - name: solarwinds/solarwinds-snap-agent-docker
-    newTag: 3.3.0-3.1.1.717
+    newTag: 3.3.1-3.1.1.717
```
Comment (lootek) [inline deploy/overlays/stable/daemonset/kustomization.yaml:11]: the version needs to be changed here accordingly

File: README.md (hunk 5 above)
Comment (melord) [reply README.md:197]: smth like this would be better I think

File: conf/swisnap-init.sh [hunk 9]
```diff
@@ -63,6 +63,39 @@ run_plugins_with_default_configs() {
         mv "${PLUGINS_DIR}/kubernetes.yaml.example" "${PLUGINS_DIR}/kubernetes.yaml"
     fi
 
+    if [ "${SWISNAP_ENABLE_NGINX}" = "true" ]; then
+        NGINX_CONFIG="${TASK_AUTOLOAD_DIR}/task-bridge-nginx.yaml"
+        mv "${NGINX_CONFIG}.example" "${NGINX_CONFIG}"
+        if [[ -n "${NGINX_STATUS_URI}" ]]; then
```
Comment (melord) [inline conf/swisnap-init.sh:69]: maybe some debug messages for users if needed vars are not set

File: Makefile (hunk 2 above)
Comment (d-maslyk) [reply Makefile:15]: As talked offline - it's reverting for test target repo from when the binaries are taken.

**Pull request merged**